marlowe-dashboard-client/src/API/Contract.purs

```diff
@@ -1,99 +1,71 @@
-  , defaultInvokeEndpoint
-  , defaultGetWalletContractInstances
-  , defaultGetAllContractInstances
-  , defaultGetContractDefinitions
   ) where
 
 import Prelude
 import API.Request (doGetRequest, doPostRequest, doPutRequest)
 import API.Url (toUrlPiece)
 import Control.Monad.Error.Class (class MonadError)
 import Effect.Aff.Class (class MonadAff)
-import Foreign.Generic (class Decode, class Encode)
-import Plutus.PAB.Effects.Contract.Builtin (Builtin)
+import Foreign.Generic (class Encode)
+import MarloweContract (MarloweContract)
 import Plutus.PAB.Webserver.Types (ContractActivationArgs, ContractInstanceClientState, ContractSignatureResponse)
 import Servant.PureScript.Ajax (AjaxError)
 import Wallet.Emulator.Wallet (Wallet)
 import Wallet.Types (ContractInstanceId)
 
--- PAB contracts can be activated with custom data type (Builtin) that
--- identifies the contract (for versions of the plutus-pab that are bundled up
--- with contracts). That value is also returned in the `ContractInstanceClientState`.
-class
-  (Decode a, Encode a) <= ContractActivationId a where
-  activateContract :: forall m. MonadError AjaxError m => MonadAff m => ContractActivationArgs a -> m ContractInstanceId
-  deactivateContract :: forall m. MonadError AjaxError m => MonadAff m => a -> ContractInstanceId -> m Unit
-  getContractInstanceClientState :: forall m. MonadError AjaxError m => MonadAff m => a -> ContractInstanceId -> m (ContractInstanceClientState a)
-  invokeEndpoint :: forall d m. MonadError AjaxError m => MonadAff m => Encode d => a -> ContractInstanceId -> String -> d -> m Unit
-  getWalletContractInstances :: forall m. MonadError AjaxError m => MonadAff m => a -> Wallet -> m (Array (ContractInstanceClientState a))
-  getAllContractInstances :: forall m. MonadError AjaxError m => MonadAff m => a -> m (Array (ContractInstanceClientState a))
-  getContractDefinitions :: forall m. MonadError AjaxError m => MonadAff m => a -> m (Array (ContractSignatureResponse a))
-
-defaultActivateContract ::
-  forall a m.
-  ContractActivationId a =>
+activateContract ::
+  forall m.
   MonadError AjaxError m =>
   MonadAff m =>
-  ContractActivationArgs a -> m ContractInstanceId
-defaultActivateContract contractActivationArgs = doPostRequest "/api/contract/activate" contractActivationArgs
+  ContractActivationArgs MarloweContract -> m ContractInstanceId
+activateContract contractActivationArgs = doPostRequest "/api/contract/activate" contractActivationArgs
 
-defaultDeactivateContract ::
-  forall a m.
-  ContractActivationId a =>
+deactivateContract ::
+  forall m.
   MonadError AjaxError m =>
   MonadAff m =>
-  a -> ContractInstanceId -> m Unit
-defaultDeactivateContract contractActivationId contractInstanceId = doPutRequest $ "/api/contract/instance/" <> toUrlPiece contractInstanceId <> "/stop"
+  ContractInstanceId -> m Unit
+deactivateContract contractInstanceId = doPutRequest $ "api/contract/instance/" <> toUrlPiece contractInstanceId <> "/stop"
```

Comment:
It seems the first slash was removed here, was this on purpouse?

Reply:
Also I have had problems with PUT requests and Servant in the past, if it gives trouble it may be worth trying a different method

Reply:
Good catch, thanks.

It's a PUT request in the PAB, so that's what we need here. So far it's working... :crossed_fingers: